run-tests.php

```diff
@@ -2959,8 +2959,34 @@ function generate_array_diff(array $ar1, array $ar2, bool $is_reg, array $w): ar
     return $diff;
 }
 
+function generate_diff_external(string $wanted, string $output): string
+{
+    global $environment;
+
+    $diff_cmd = $environment['DIFF_CMD'];
+
+    /* Write two temp files */
+    $wanted_file = $environment['TEMP'] . DIRECTORY_SEPARATOR . 'wanted.txt';
+    $output_file = $environment['TEMP'] . DIRECTORY_SEPARATOR . 'output.txt';
+    file_put_contents($wanted_file, $wanted);
+    file_put_contents($output_file, $output);
+
+    $diff_output = shell_exec("{$diff_cmd} $wanted_file $output_file");
+
+    unlink($wanted_file);
+    unlink($output_file);
+
+    return $diff_output;
+}
+
 function generate_diff(string $wanted, ?string $wanted_re, string $output): string
 {
+    global $environment;
+
+    if (!empty($environment['DIFF_CMD'])) {
```

Comment:
Is this a standardized variable? If not maybe there should be some PHP prefix?

Reply:
Updated too - there is no standardized one it seems.